developer/src/kmc/src/commands/build.ts

```diff
@@ -102,7 +102,7 @@ async function buildFile(filenames: string[], _options: any, commander: any)  {
   }
 
   for(let filename of filenames) {
-    if(!await build(filename, commanderOptions.outFile, callbacks, options)) {
+    if(!await build(filename, commanderOptions.outFile, callbacks, options) || callbacks.hasFailureMessage()) {
```

Comment:
I'm not sure about this. It should probably be an assertion. We should never return `true` from `build()` if we have an error that fails the build. Feels like if we do, we have a bug in the compiler.

Reply:
I could make it an assertion.

Reply:
debug build assertion sounds appropriate (or log to sentry)

Reply:
added a `kmcAssert()` that throws an internal error if falsy.

Reply:
Given we have the following in `build()`:

https://github.com/keymanapp/keyman/blob/898310e6df05d07c6ddb068fdc96e5489e46b5dd/developer/src/kmc/src/commands/build.ts#L158-L159

This is already being handled by the build() function, and we should keep that logic there.

```suggestion
    if(!await build(filename, commanderOptions.outFile, callbacks, options)) {
```